### PR TITLE
Cyhy dashboard

### PR DESCRIPTION
**Pull request by KyleEvers**

### CODE REVIEW

**Comment by dav3r** [comment]
I gave some verbal comments to @KyleEvers and he is working on the changes.

File: ansible/roles/cyhy_dashboard/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,114 @@
+---
+# tasks file for cyhy_dashboard
+
+#
```
Comment (dav3r) [inline ansible/roles/cyhy_dashboard/tasks/main.yml]: These three comment lines can be removed or updated to more accurately reflect what is happening here

File: ansible/roles/cyhy_dashboard/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      [DEFAULT]
+      default-section = production
+      database-uri = mongodb://localhost:27017/
+      report-key =
+
+      [production]
+      database-uri = mongodb://commander:{{ mongo_users.other_users | selectattr("user", "eq", "commander") | map(attribute="password") | first }}@database1:27017/cyhy
+      database-name = cyhy
+
+- name: Start ncat-webd
```
Comment (dav3r) [inline ansible/roles/cyhy_dashboard/tasks/main.yml]: Nitpick: This should read: "Start ncat**s**-webd'"

File: packer/ansible/roles/cyhy_dashboard/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
+---
+# tasks file for cyhy_dashboard
+
+#
```
Comment (dav3r) [inline packer/ansible/roles/cyhy_dashboard/tasks/main.yml]: These 3 comment lines can be removed.

File: packer/ansible/roles/cyhy_dashboard/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+# Install curl and unzip; they are needed by cyhy_core/var/load_places.sh
+#
+- name: Install pip
```
Comment (dav3r) [inline packer/ansible/roles/cyhy_dashboard/tasks/main.yml]: We already have a `pip` role at `packer/ansible/roles/pip`.  This section should be removed and `pip` should be added to the list of dependencies in `packer/ansible/roles/cyhy_dashboard/meta/main.yml`

**Review by dav3r — CHANGES_REQUESTED** [review]
Just needs a few minor changes @KyleEvers and then this should be good to go

File: ansible/roles/cyhy_dashboard/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    group: cyhy
+    mode: 0750
+
+- name: Create /var/cyhy directory
+  file:
+    path: /var/cyhy
+    state: directory
+    owner: cyhy
+    group: cyhy
+    mode: 0750
```
Comment (jsf9k) [inline ansible/roles/cyhy_dashboard/tasks/main.yml:33]: Is there a reason why `/var/cyhy` and `/var/cyhy/web` need different modes?

File: ansible/roles/cyhy_dashboard/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+      [DEFAULT]
+      default-section = production
+      database-uri = mongodb://localhost:27017/
+      report-key =
+
+      [production]
+      database-uri = mongodb://commander:{{ mongo_users.other_users | selectattr("user", "eq", "commander") | map(attribute="password") | first }}@database1:27017/cyhy
+      database-name = cyhy
+
+- name: Start ncats-webd
```
Comment (jsf9k) [inline ansible/roles/cyhy_dashboard/tasks/main.yml]: This isn't a good way to start a service.  If the instance reboots the service won't be restarted.  You should really create a systemd unit instead.

Look at the packer and terraform ansibles for the commander for an example.

File: ansible/roles/cyhy_dashboard/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+  shell: ncats-webd -ls production &
+  become_method: sudo
+  become_user: cyhy
+  vars:
+    ansible_ssh_pipelining: yes
+
+# Run `dev-build` target as root
```
Comment (jsf9k) [inline ansible/roles/cyhy_dashboard/tasks/main.yml]: This is nit-picky, but "Run" is incorrect here.  It should be "Build".

File: ansible/roles/cyhy_dashboard/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+- make:
+    chdir: /var/cyhy/ncats-webui
+    target: dev-build
+  become: yes
+
+# Run `dev-start` target as root
+- make:
```
Comment (jsf9k) [inline ansible/roles/cyhy_dashboard/tasks/main.yml]: Again, this is a terrible way to start a service.  If the instance reboots the service won't be restarted.  You should really create a systemd unit instead.

File: packer/ansible/roles/cyhy_dashboard/README.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+------------
+
+None
+
+Example Playbook
+----------------
+
+Here's how to use it in a playbook:
+
+     - hosts: cyhy_dashboard
```
Comment (jsf9k) [inline packer/ansible/roles/cyhy_dashboard/README.md:26]: Bad formatting here.  The lines that follow are not sufficiently indented.

File: ansible/roles/cyhy_dashboard/meta/main.yml [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+    # and categorizes the role. Users find roles by searching for tags. Be sure to
+    # remove the '[]' above, if you add tags to this list.
+    #
+    # NOTE: A tag is limited to a single word comprised of alphanumeric characters.
+    #       Maximum 20 tags per role.
+
+dependencies:
+  # List your role dependencies here, one per line. Be sure to remove the '[]' above,
+  # if you add dependencies to this list.
+  - github_oauth
```
Comment (jsf9k) [inline ansible/roles/cyhy_dashboard/meta/main.yml]: I don't see where the token from `github_oauth` is used here.

File: packer/ansible/roles/cyhy_dashboard/tasks/main.yml [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+    src: "https://api.github.com/repos/jsf9k/ncats-webd/tarball/develop?access_token={{ github_oauth_token }}"
+    dest: /tmp/ncats-webd
+    remote_src: yes
+    extra_opts:
+      - "--strip-components=1"
+
+- name: Install webd
+  command: pip install /tmp/ncats-webd
+
+- name: Create /var/cyhy/webui directory
```
Comment (jsf9k) [inline packer/ansible/roles/cyhy_dashboard/tasks/main.yml]: The directory name is incorrect in this comment.

File: terraform/cyhy_dashboard_ec2.tf [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  owners = ["${data.aws_caller_identity.current.account_id}"] # This is us
+  most_recent = true
+}
+
+# The cyhy dashboard EC2 instance
+resource "aws_instance" "cyhy_dashboard" {
+  ami = "${data.aws_ami.dashboard.id}"
+  instance_type = "${local.production_workspace ? "t3.medium" : "t3.medium"}"
```
Comment (jsf9k) [inline terraform/cyhy_dashboard_ec2.tf]: No need for a conditional if the result is the same either way.

**Review by jsf9k — CHANGES_REQUESTED** [review]
The systemd unit stuff isn't necessary for approval, but if not it should be fixed in a future pull request.

Please see my other comments as well.

**Review by dav3r — APPROVED** [review]

**Review by jsf9k — APPROVED** [review]

**Pull request merged**